### ISSUE TITLE
`logger:notice('~p', [a])` removes default filter

### ISSUE THREAD

**Issue by hauleth** [erlang/otp#5071]
**Describe the bug**

Using `logger:log(Level, '~p', [a])` results in message that default filter was removed:

```
Logger - error: {removed_failing_filter,no_domain}
```

**To Reproduce**

Run `logger:info('~p', [a])` in shell

**Expected behavior**

There should be no failure in filter and message `a` should be logged.

**Affected versions**

Tested on OTP 24, but I assume that all versions since OTP 21 are affected.

**Additional context**

I wanted to check whether `logger:info(report, [a])` will behave as expected as `logger:log_event()` defines `msg` field as `{io:format(), term()} | {report, logger:report()} | {string, unicode:chardata()}`, however as `io:format()` can also be an atom, it may cause issues when pattern matching against `{report, Report}` or `{string, Msg}`.

**Comment by lhoguin** [comment]
I do not get the `logger` error when I try in the shell (`[erts-12.0.2]`).

**Comment by hauleth** [comment]
I am using ERTS 12.0.3 and:

- `logger:notice(string, [a]).` prints:
   ```
   =NOTICE REPORT==== 27-Jul-2021::12:41:19.046491 ===
   FORMAT ERROR: "~ts" - [[a]]
   ```
- `logger:notice('~p', [a]).` prints: `Logger - error: {removed_failing_filter,no_domain}`
- `logger:notice(report, [a]).` prints: `Logger - error: {removed_failing_filter,no_domain}`

I have tested on ERTS 11.2.2.3, and behaviour is exactly the same.

**Comment by lhoguin** [comment]
Right I get it with notice but not with info, I guess the "To Reproduce" section should be updated.

**Comment by hauleth** [comment]
@lhoguin yes, it always trip me that Erlang uses `notice` as a default verbosity level. Sorry for the confusion.
